src/plugins/intel_cpu/src/compiled_model.h

```diff
@@ -94,5 +100,60 @@ class CompiledModel : public ov::ICompiledModel {
     bool m_has_sub_compiled_models = false;
 };
 
+// This class provides safe access to the internal CompiledModel structures and helps to decouple SyncInferRequest and
+// the CompiledModel internal structures
+class CompiledModelHolder {
+public:
+    CompiledModelHolder(std::shared_ptr<const CompiledModel> compiled_model)
+        : m_compiled_model(std::move(compiled_model)) {
+        OPENVINO_ASSERT(!m_compiled_model->m_graphs.empty(),
+                        "No graph was found in the compiled model: ",
+                        m_compiled_model->name());
+        m_graph = &(m_compiled_model->get_graph()._graph);
+        OPENVINO_ASSERT(m_graph, "Graph ptr null check failed");
+        m_id = (m_compiled_model->m_numRequests)++;
```

Comment:
It is not clear why do we increase a numRequests in scope of this object constructor.
I mean this object is not related to the InferRequest object at all.
Probably the more proper question is - what is the purpose of this counter.
I would understand if this was a unique id of an infer request.
But from my understanding, if one creates and infer requests, then deletes the object, and then creates a new infer request, the second infer request will get the same id which is a bit confusing. This helps with a possible overflow but does not really make sense.
Maybe I am missing something

Reply:
> Probably the more proper question is - what is the purpose of this counter.

The only usage of this id - itt performance counters. Probably it's been done to reduce the amount of INTEL_CPU_INFER_ records in the itt perf counter report for use cases where by some reason an infer request is created and destroyed in a loop.